modules/sources/enrichers/get-liquidity-at-timestamp.ts

```diff
@@ -0,0 +1,64 @@
+import { BlockNumbersClient, V3VaultSubgraphClient } from '../subgraphs';
+import { V2VaultSubgraphClient } from '../../subgraphs/balancer-subgraph';
+import { prisma } from '../../../prisma/prisma-client';
+import { daysAgo, roundToHour, roundToMidnight } from '../../common/time';
+import { fn } from '../../common/numbers';
+import { DAYS_OF_HOURLY_PRICES } from '../../../config';
+
+export const getLiquidityAtTimestamp = async (
+    ids: string[],
+    vaultClient: V2VaultSubgraphClient | V3VaultSubgraphClient,
+    blockNumbersClient: BlockNumbersClient,
+    timestamp = daysAgo(1), // 24 hours ago
+) => {
+    const blockNumber = await blockNumbersClient.fetchBlockByTime(timestamp);
+
+    //  If ids count is >= 1000 just get all
+    const where = ids.length >= 1000 ? {} : { id_in: ids };
+    const balances = await vaultClient.getAllPoolBalances({
+        where,
+        block: { number: Number(blockNumber) },
+    });
+
+    // Guard against empty pools, for example when the pools weren't created yet
+    if (!balances.length) {
```

Comment:
shouldn't that be lenght === 0? Can it be undefined?

Reply:
either way